modules/utxo-lib/src/bitgo/UtxoPsbt.ts

```diff
@@ -322,6 +327,37 @@ export class UtxoPsbt<Tx extends UtxoTransaction<bigint> = UtxoTransaction<bigin
     return this;
   }
 
+  /**
+   * Finalizes a taproot musig2 input by aggregating all partial sigs.
+   * IMPORTANT: Always call validate* function before finalizing.
```

Comment:
maybe we should do it inside this method then?

what happens if the caller forgets about it?

Reply:
Just following the [bitcoinjs-lib](https://github.com/bitcoinjs/bitcoinjs-lib/blob/master/ts_src/psbt.ts#L114) Psbt.ts standard flow which is consistence across other input types.

The validate* function checks script, pub keys, sigs and other meta data completely.
Finalize* is a dumb function. Just moves everything to different fields, so later extraction of network tx will be quicker, cheaper and easier.

If the caller doesn't ensure validity with validate* function, there are different possible outcomes:
1. Finalize will be successful if all the signatures exist (but it can be wrong signature or pub keys too)
2. Finalize will throw error if invalid number of sigs exist or few metadata like tapInternalKey is missing.

Upstream requires changes to make the requested behaviour consistent across inputs.